frontend/app/routes/protected/person-case/primary-docs.tsx

```diff
@@ -46,37 +57,64 @@ export async function action({ context, request }: Route.ActionArgs) {
+      throw i18nRedirect('routes/protected/person-case/request-details.tsx', request);
+    }
 
-  // submit action
-  const schema = v.object({
-    currentStatusInCanada: v.pipe(
-      v.string(t('protected:primary-identity-document.current-status-in-canada.required')),
-      v.trim(),
-      v.nonEmpty(t('protected:primary-identity-document.current-status-in-canada.required')),
-      v.literal(
-        'canadian-citizen-born-outside-canada',
-        t('protected:primary-identity-document.current-status-in-canada.invalid'),
-      ),
-    ),
-  });
-
-  const input = { currentStatusInCanada: formData.get('currentStatusInCanada') as string };
-  const parsedDataResult = v.safeParse(schema, input, { lang });
-
-  if (!parsedDataResult.success) {
-    return data({ errors: v.flatten<typeof schema>(parsedDataResult.issues).nested }, { status: 400 });
-  }
+    case 'next': {
+      const schema = v.intersect([
+        v.object({
+          currentStatusInCanada: v.pipe(
+            v.string(t('protected:primary-identity-document.current-status-in-canada.required')),
+            v.trim(),
+            v.nonEmpty(t('protected:primary-identity-document.current-status-in-canada.required')),
+            v.picklist(VALID_CURRENT_STATUS, t('protected:primary-identity-document.current-status-in-canada.invalid')),
+          ),
+        }),
+        v.variant(
+          'documentTypeAvailable',
+          [
+            v.object({
+              documentTypeAvailable: v.literal(false),
+              documentType: v.optional(v.string()),
+            }),
+            v.object({
+              documentTypeAvailable: v.literal(true),
+              documentType: v.pipe(v.picklist(VALID_DOCTYPE, t('protected:primary-identity-document.document-type.required'))),
+            }),
+          ],
+          t('protected:primary-identity-document.document-type.required'),
+        ),
+      ]) satisfies v.GenericSchema<PrimaryDocumentsSessionData>;
 
-  context.session.inPersonSINCase = {
-    ...(context.session.inPersonSINCase ?? {}),
-    ...input,
-  };
+      const input = {
+        currentStatusInCanada: formData.get('currentStatusInCanada') as string,
+        documentType: documentTypeAvailable ? (formData.get('documentType') as string) : undefined,
+        documentTypeAvailable: documentTypeAvailable,
+      } satisfies Partial<PrimaryDocumentsSessionData>;
+
+      const parseResult = v.safeParse(schema, input, { lang });
+
+      if (!parseResult.success) {
+        return data({ errors: v.flatten<typeof schema>(parseResult.issues).nested }, { status: 400 });
+      }
 
-  throw i18nRedirect('routes/protected/request.tsx', request); //TODO: change it to redirect to file="routes/protected/person-case/secondary-docs.tsx"
+      context.session.inPersonSINCase ??= {};
+      context.session.inPersonSINCase.primaryDocuments = {
+        ...parseResult.output,
+        documentType: parseResult.output.documentType ?? '',
+      };
```

Comment:
Entire validated validation should be stored as is.

```suggestion
      context.session.inPersonSINCase.primaryDocuments = parseResult.output;
```